File: python/fundamentals/day-06/MiniChallenge.py
marks = [85, 72, 91, 68, 88]

total = 0
highest = marks[0]
lowest = marks[0]
passed = 0
failed = 0

# Properly indented loop body
for mark in marks:
    total += mark

    # Track highest mark
    if mark > highest:
        highest = mark

    # Track lowest mark
    if mark < lowest:
        lowest = mark

    # Pass/Fail evaluation (Passing mark = 35)
    if mark >= 35:
        passed += 1
    else:
        failed += 1

average = total / len(marks)
result = "PASS" if failed == 0 else "FAIL"

# Formatted Output
print("==============================")
print("       MARKS ANALYZER        ")
print("==============================")
print("Marks   :", *marks)
print(f"Total   : {total}")
print(f"Average : {average:.2f}")
print(f"Highest : {highest}")
print(f"Lowest  : {lowest}")
print(f"Passed  : {passed}")
print(f"Failed  : {failed}")
print(f"Result  : {result}")
print("==============================")

File: java/fundamentals/day06/MiniChallenge.java
package fundamentals.day06;

public class MiniChallenge {

	    public static void main(String[] args) {
	        int[] marks = {85, 72, 91, 68, 88};

	        int total = 0;
	        int highest = marks[0];
	        int lowest = marks[0];
	        int passed = 0;
	        int failed = 0;

	        // Loop to calculate all metrics in a single pass
	        for (int mark : marks) {
	            total += mark;

	            // Track highest mark
	            if (mark > highest) {
	                highest = mark;
	            }

	            // Track lowest mark
	            if (mark < lowest) {
	                lowest = mark;
	            }

	            // Pass/Fail evaluation (Passing mark = 35)
	            if (mark >= 35) {
	                passed++;
	            } else {
	                failed++;
	            }
	        }

	        double average = (double) total / marks.length;
	        String result = (failed == 0) ? "PASS" : "FAIL";

	        // Formatted Output
	        System.out.println("==============================");
	        System.out.println("       MARKS ANALYZER        ");
	        System.out.println("==============================");
	        
	        System.out.print("Marks   : ");
	        for (int i = 0; i < marks.length; i++) {
	            System.out.print(marks[i] + (i < marks.length - 1 ? " " : ""));
	        }
	        System.out.println();

	        System.out.println("Total   : " + total);
	        System.out.printf("Average : %.2f%n", average);
	        System.out.println("Highest : " + highest);
	        System.out.println("Lowest  : " + lowest);
	        System.out.println("Passed  : " + passed);
	        System.out.println("Failed  : " + failed);
	        System.out.println("Result  : " + result);
	        System.out.println("==============================");
	    }
	}
